### PR TITLE
Add some initial api tests for budgets and accounts

### PR DESCRIPTION
**Pull request by twk3**
- Adds jest config for the api package
- Reuse setup from loot-core, as we are loading it's server/main anyways
- Cover some basic budget and account tests to prove it out

The test setup is based on what was done for server/main.test.ts (and reuses it's mock databaseFiles)

Part of https://github.com/actualbudget/actual/issues/1074

### CODE REVIEW

File: packages/api/tsconfig.dist.json [hunk 1]
```diff
@@ -11,5 +11,5 @@
     "outDir": "dist"
   },
   "include": ["."],
-  "exclude": ["dist"]
+  "exclude": ["dist", "**/*.test.js"]
```
Comment (MatissJanis) [inline packages/api/tsconfig.dist.json:14]: **💬 suggestion:** ‏ideally we shouldn't exclude test files. Also: the test files should be `.ts` not `.js`.

Comment (twk3) [reply packages/api/tsconfig.dist.json]: It looks like typecheck still runs. But I still took it out as it wasn't needed after the other changes you requested, and the other packages aren't excluding the tests at this time.

**Comment by twk3** [comment]
@MatissJanis I addressed your feedback. Everything went pretty smooth, I added the mocks/budget to the watch ignore so it shouldn't loop.

Other than what you requested (typescript, drop the loot-core imports), the main other thing I had to do was have the test script build the api prior to testing. Otherwise we don't have a bundle for the index to import.

**Review by MatissJanis — APPROVED** [review]
Amazing! Thanks for your hard work on the API tests. This will definitely make things more stable in the future.

**Pull request merged**